templates/commands/render_test.go

```diff
@@ -155,11 +160,177 @@ func TestDestOK(t *testing.T) {
+
+			tempDir := t.TempDir()
+			templateDir := filepath.Join(tempDir, "template")
+			templateDirNamer := func(debugID string) (string, error) {
+				return templateDir, nil
+			}
+			rp := &runParams{
+				getter: &fakeGetter{
+					wantSource: "github.com/myorg/myrepo",
+					output:     tc.templateContents,
+					err:        tc.getterErr,
+				},
+				fs: &errorFS{
+					renderFS:     &realFS{},
+					removeAllErr: tc.removeAllErr,
+				},
+				tempDirNamer: templateDirNamer,
+			}
+			r := &Render{
+				source:           "github.com/myorg/myrepo",
+				flagKeepTempDirs: tc.keepTempDirs,
+			}
+			ctx := context.Background()
+			err := r.realRun(ctx, rp)
+			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+				t.Error(diff)
+			}
+
+			wantFiles := tc.templateContents
+			if !tc.keepTempDirs {
+				wantFiles = nil
+			}
+
+			got := loadDirContents(t, templateDir)
+			if diff := cmp.Diff(got, wantFiles); diff != "" {
+				t.Errorf("output files were not as expected (-got,+want): %s", diff)
+			}
+		})
+	}
+}
+
+// Read all the files recursively under "dir", returning their contents as a
+// map[filename]->contents. Returns nil if dir doesn't exist.
+func loadDirContents(t *testing.T, dir string) map[string]string {
+	t.Helper()
+	out := map[string]string{}
+	if _, err := os.Stat(dir); err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil
+		}
+		t.Fatal(err)
+	}
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		contents, err := os.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("ReadFile(): %w", err)
+		}
+		rel, err := filepath.Rel(dir, path)
+		if err != nil {
+			return fmt.Errorf("Rel(): %w", err)
+		}
+		out[rel] = string(contents)
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("WalkDir(): %v", err)
+	}
+	return out
+}
+
+// A renderFS implementation that can inject errors for testing.
 type errorFS struct {
-	fs.StatFS
-	err error
+	renderFS
+
+	statErr      error
+	removeAllErr error
+	openErr      error
+}
+
+func (e *errorFS) Stat(name string) (fs.FileInfo, error) {
+	if e.statErr != nil {
+		return nil, e.statErr
+	}
+	return e.renderFS.Stat(name)
 }
 
-func (e *errorFS) Stat(string) (fs.FileInfo, error) {
-	return nil, e.err
+func (e *errorFS) RemoveAll(name string) error {
+	if e.removeAllErr != nil {
+		return e.removeAllErr
+	}
+	return e.renderFS.RemoveAll(name)
+}
+
+func (e *errorFS) Open(name string) (fs.File, error) {
+	if e.openErr != nil {
+		return nil, e.openErr
+	}
+	return e.renderFS.Open(name)
+}
+
+type fakeGetter struct {
+	wantSource string
+	output     map[string]string
+	err        error
+}
+
+func (f *fakeGetter) Get(ctx context.Context, req *getter.Request) (*getter.GetResult, error) {
+	if f.err != nil {
+		return nil, f.err
+	}
+	if req.Src != f.wantSource {
+		return nil, fmt.Errorf("got template source %s but expected %s", req.Src, f.wantSource)
+	}
```

Comment:
I'd prefer comparing got/want in test instead since it's more explicit.

Reply:
I think you're looking for the word "expected" to be "want" instead? I made that change.